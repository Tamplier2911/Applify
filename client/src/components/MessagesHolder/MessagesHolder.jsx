// import "./MessagesHolder.scss";
import React, { useEffect, useState } from "react";

// redux
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectCurrentLanguage } from "../../redux/lang/lang.selectors";
import { loadMessagesStart } from "../../redux/messages/messages.actions";
import {
  selectAllMessages,
  selectIsLoading,
} from "../../redux/messages/messages.selectors";

// components
import WithSpinnerHOC from "../../components/WithSpinnerHOC/WithSpinnerHOC";
import CollectionHolder from "../CollectionHolder/CollectionHolder";
import MessagesCollection from "../MessagesCollection/MessagesCollection";
import FormInput from "../FormInput/FormInput";

// data transformations
import { simpleMessageSearch } from "../../utils/DataTransformations/simpleSearches";

// JS Rendering CSS
import {
  MessagesHolderContainer,
  MessageHolderInputBarContainer,
} from "./MessagesHolderStyles";

// component constants
import messagesHolderData from "./MessagesHolderConstants";

// Buffing MessagesCollection with Spinner
const MessagesCollectionWithSpinner = WithSpinnerHOC(MessagesCollection);

const MessagesHolder = ({ lang, loadMessagesStart, messages, isLoading }) => {
  useEffect(() => {
    if (!messages.length) {
      loadMessagesStart();
    }
  }, [loadMessagesStart, messages.length]);

  const [searchInput, setSearchInput] = useState({ search: "" });
  const { search } = searchInput;

  const onInputChange = (e) => {
    const { name, value } = e.target;
    setSearchInput({ ...searchInput, [name]: value });
  };

  const messagesArray = simpleMessageSearch(messages, search.toLowerCase());

  const { messagesHolderTitle, messageHolderSearch } = messagesHolderData[lang];
  return (
    <MessagesHolderContainer data-test="messages-holder">
      <MessageHolderInputBarContainer data-test="messages-holder-search">
        <FormInput
          onInputChange={(e) => onInputChange(e)}
          value={search}
          name="search"
          label={messageHolderSearch}
          type="text"
        />
      </MessageHolderInputBarContainer>
      <CollectionHolder
        lang={lang}
        title={messagesHolderTitle}
        refresher={1}
        cb={loadMessagesStart}
        data-test="messages-holder-collection-holder"
      >
        <MessagesCollectionWithSpinner
          data-test="messages-holder-collection"
          isLoading={isLoading}
          messages={messagesArray}
        />
      </CollectionHolder>
    </MessagesHolderContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  lang: selectCurrentLanguage,
  messages: selectAllMessages,
  isLoading: selectIsLoading,
});

export default connect(mapStateToProps, { loadMessagesStart })(MessagesHolder);
