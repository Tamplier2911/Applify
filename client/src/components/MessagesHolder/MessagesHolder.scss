/*

.messages-holder {
  grid-row: 1/2;
  animation: 1s holderFadeIn;
  display: grid;

  @media only screen and (max-width: 768px) {
    margin-bottom: 5rem;
  }

  &__title {
    //   ${getFont}
    font-family: var(--font-title);
    font-size: 4rem;
    font-weight: 300;
    padding: 1rem;

    @media only screen and (max-width: 768px) {
      line-height: 1;
    }
  }

  &__messages {
    display: grid;
  }
}

@keyframes holderFadeIn {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

*/
