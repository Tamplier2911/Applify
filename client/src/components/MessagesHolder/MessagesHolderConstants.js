const messagesHolderData = {
  eng: {
    messagesHolderTitle: "Recent Messages:",
    messageHolderSearch: "Search:",
  },
  rus: {
    messagesHolderTitle: "Последние сообщения:",
    messageHolderSearch: "Поиск:",
  },
  ukr: {
    messagesHolderTitle: "Останні повідомлення:",
    messageHolderSearch: "Пошук:",
  },
};

export default messagesHolderData;
