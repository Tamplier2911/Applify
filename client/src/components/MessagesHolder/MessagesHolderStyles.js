import styled from "styled-components";

export const MessagesHolderContainer = styled.div``;
