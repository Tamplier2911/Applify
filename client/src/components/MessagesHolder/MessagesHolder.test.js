import React from "react";
import toJson from "enzyme-to-json";

// components
import MessagesHolder from "./MessagesHolder";

// utils
import {
  setShallowC,
  findByTestAttr,
  storeFactory,
} from "../../tests/testUtils";

describe("<MessagesHolder />", () => {
  const initialStore = {
    lang: {
      languagePanelHidden: true,
      currentLanguage: "eng",
    },
    messages: {
      allMessages: {},
      isLoading: false,
      errorMessage: null,
    },
  };
  const store = storeFactory(initialStore);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowC(<MessagesHolder store={store} />);
  });

  it("renders without an error", () => {
    const messagesHolder = findByTestAttr(wrapper, "messages-holder");
    expect(messagesHolder.length).toBe(1);
  });

  it("renders children - search bar", () => {
    const messagesHolderSearch = findByTestAttr(
      wrapper,
      "messages-holder-search"
    );
    expect(messagesHolderSearch.length).toBe(1);
  });

  it("renders children - collection holder", () => {
    const messagesHolderColHolder = findByTestAttr(
      wrapper,
      "messages-holder-collection-holder"
    );
    expect(messagesHolderColHolder.length).toBe(1);
  });

  it("renders children - collection", () => {
    const messagesHolderCollection = findByTestAttr(
      wrapper,
      "messages-holder-collection"
    );
    expect(messagesHolderCollection.length).toBe(1);
  });

  it("matches snapshot", () => {
    const messagesHolder = findByTestAttr(wrapper, "messages-holder");
    expect(toJson(messagesHolder)).toMatchSnapshot();
  });
});
