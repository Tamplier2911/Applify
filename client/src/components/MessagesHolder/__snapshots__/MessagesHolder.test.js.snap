// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessagesHolder /> matches snapshot 1`] = `ShallowWrapper {}`;
